src/android/toga_android/widgets/numberinput.py

```diff
@@ -60,12 +72,12 @@ def set_alignment(self, value):
     def set_font(self, value):
         self.interface.factory.not_implemented("NumberInput.set_font()")
 
-    def get_value(self):
-        return Decimal(self.native.getText())
-
     def set_value(self, value):
-        # Toga's `value` is a `Decimal`, but Android needs a string.
-        self.native.setText(str(value))
+        # Toga's `value` is a `Decimal` or `None`, but Android needs a string.
+        str_value = str(value) if value is not None else ""
+        if str_value == self.native.getText().toString():
+            return
```

Comment:
Is there a performance benefit to this shortcut? I would have thought the API round trip and comparison would have been more expensive than just setting the same value.

Reply:
The reason it's here is that at one point I found myself getting a recursion depth exceeded error from Python. I think (but didn't 100% verify) that the cause is that `self.value = ...` ends up triggering this code, which then blindly overwrites `self.value - ```, which is an infinite loop.

I could cache it not in the native widget to avoid that infinite loop. Open to other ideas, including you asking me to verify this. :)

Reply:
Ah - makes sense. So: If you can verify that this is actually the case, that would be great. If it turns out it is needed, then adding a comment explaining why it exists will prevent someone... optimizing... in future.

Reply:
FYI: I've just done some testing, and from what I can tell, it's not needed.

Reply:
I believe this is fixed now that I do `self.interface._value = ...` not `self.interface.value = ...`. Marking resolved for now.